src/engine/source/api/src/kvdb/handlers.cpp

```diff
@@ -71,19 +72,42 @@ api::Handler managerPost(std::shared_ptr<kvdb_manager::KVDBManager> kvdbManager)
             return ::api::adapter::genericError<ResponseType>("Missing /name");
         }
 
-        const auto path = eRequest.has_path() ? eRequest.path() : std::string {""};
-        auto result = kvdbManager->createFromJFile(eRequest.name(), path);
-        if (result.has_value())
+        if (std::empty(eRequest.name()))
         {
-            return ::api::adapter::genericError<ResponseType>(result.value().message);
+            return ::api::adapter::genericError<ResponseType>("/name is empty");
+        }
+
+        auto resultExists = kvdbManager->existsDB(eRequest.name());
+
+        if (resultExists)
+        {
+            return ::api::adapter::genericError<ResponseType>("The Database already exists.");
+        }
+
+        auto resultCreate = kvdbManager->createDB(eRequest.name());
+
+        if (resultCreate)
+        {
+            return ::api::adapter::genericError<ResponseType>(resultCreate.value().message);
+        }
+
+        if (eRequest.has_path())
+        {
+            auto resultLoad = kvdbManager->loadDBFromFile(eRequest.name(), eRequest.path());
+
+            if (resultLoad)
+            {
+                std::string message = fmt::format("The DB was created but loading data returned: {}", resultLoad.value().message);
```

Comment:
I think that the proposed message does not make it clear if the DB was created or not, what do you think?

Reply:
Please disregard this. It's a temporal message because it's part of a pending feature. This message just states what happened. But to keep consistency we used that message.